src/python/move_emojis/scripts/generate_code.py

```diff
@@ -0,0 +1,146 @@
+            elif char in [":", "-", ",", " "]:
+                sanitized_name += "_"
+            elif char == "#":
+                sanitized_name += "POUND"
+            elif char == "*":
+                sanitized_name += "ASTERISK"
+
+        words = []
+        if " " in sanitized_name:
+            words = sanitized_name.split(" ")
+        else:
+            words = [sanitized_name]
+
+        alnum_words = filter(lambda x: x.replace("_", "").isalnum(), words)
+        const_name = "_".join([w.upper() for w in alnum_words])
+
+        # Get rid of multiple underscores.
+        split_underscore = filter(lambda x: len(x) > 0, const_name.split("_"))
+        const_name = "_".join(split_underscore)
+
+        if const_name and const_name[0] in string.digits:
+            const_name = f"EMOJI_{const_name}"
+        if len(const_name) == 0:
+            print(f"ERROR: invalid constant name: {name} => {const_name}")
+            sys.exit(1)
+        move_string = "".join(data["code_points"]["as_hex"])
+        if const_name in consts:
+            print(f"ERROR: duplicate constant name: {const_name}")
+            sys.exit(1)
+        const_name = unidecode(const_name)
+        consts[const_name] = move_string
+
+    return consts
+
+
+def generate_move_code(viable_emojis: dict[str, EmojiData]) -> str:
+    # Map the full hex string to its original unicode code point.
+    original_unicode_points: dict[str, list[str]] = {}
+    for data in viable_emojis.values():
+        hex_str = "".join(data["code_points"]["as_hex"])
+        original_unicode_points[hex_str] = data["code_points"]["as_unicode"]
+
+    consts = to_human_readable(viable_emojis)
+    consts = dict(list(sorted(consts.items(), key=lambda x: x[0])))
+    consts_reverse_dict = {v: k for k, v in consts.items()}
+    hex_strings = list(consts.values())
+
+    return_open = f"{TAB*2}vector<vector<u8>> ["
+    # The hex bytes args, the aka vector<vector<u8>> args.
+    args_and_comments: list[tuple[str, str]] = []
+    for fhs in hex_strings:
+        # Lookup the const name by its full hex string and split it into
+        # words by the underscore.
+        name_split = consts_reverse_dict[fhs].split("_")
+        # Convert the name to lowercase, then capitalize the first letter.
+        name = " ".join([s.lower() for s in name_split]).capitalize()
+        original_unicode = original_unicode_points[fhs]
+        comment = f' // {name}, {" ".join(original_unicode)}.'
+        hex_value = f'x"{fhs}"'
+        arg = f"{TAB*3}{hex_value}, "
+        args_and_comments.append((arg, comment))
+
+    # Size all lines of code (up to the comment) to the same length.
+    longest_arg = max(len(a) for a, _ in args_and_comments)
+    full_args: list[str] = []
+    for arg, comment in args_and_comments:
+        full_args.append(f"{arg:<{longest_arg}}{comment}")
+    return_close = f"{TAB*2}]"
+
+    # Generate the zeroed hex string. This results in a vector of vectors,
```

Comment:
Reminder to fix this comment, not a big deal but it's a single vector `vector<u8>`, not a vector of vectors like `vector<vector<u8>>`

Reply:
Captured as checklist item in ECO-1557

Reply:
Fixed in https://github.com/econia-labs/emojicoin-dot-fun/pull/10